internal/services/instance/ip_test.go

```diff
@@ -308,8 +308,8 @@ func isIPAttachedToServer(tt *acctest.TestTools, ipResource, serverResource stri
 			return err
 		}
 
-		if server.Server.PublicIP.Address.String() != ip.IP.Address.String() {
-			return fmt.Errorf("IPs should be the same in %s and %s: %v is different than %v", ipResource, serverResource, server.Server.PublicIP.Address, ip.IP.Address)
+		if server.Server.PublicIP != nil && server.Server.PublicIP.Address.String() != ip.IP.Address.String() { //nolint:staticcheck
```

Comment:
We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.